core/src/test/java/com/gentics/mesh/core/ssl/SSLTestClient.java

```diff
@@ -0,0 +1,159 @@
+	public static final String CA_CERT = "src/test/resources/client-ssl/server.pem";
+	public static final String FMT_TEST_URL = "https://localhost:%s/api/v1";
+
+	/**
+	 * Invoke call to /api/v1
+	 * 
+	 * @param httpsPort
+	 * @param sendClientAuth
+	 * @param trustAll
+	 * @throws IOException
+	 */
+	public static void call(int httpsPort, boolean sendClientAuth, boolean trustAll) throws IOException {
+		OkHttpClient client = client(sendClientAuth, trustAll);
+		Request request = new Request.Builder().url(String.format(FMT_TEST_URL, httpsPort)).build();
+
+		log.info("Performing request: " + request);
+		Response response = client.newCall(request).execute();
+		log.info("Received response: " + response);
+	}
+
+	public static OkHttpClient client(boolean sendClientAuth, boolean trustAll) {
+		KeyManager[] keyManagers = null;
+		TrustManager[] trustManagers = null;
+
+		if (sendClientAuth) {
+			try {
+				log.info("Loading private key from " + CLIENT_KEY_PEM);
+				keyManagers = getKeyManagersPem();
+			} catch (KeyStoreException | UnrecoverableKeyException | NoSuchAlgorithmException | IOException | CertificateException e) {
+				rethrow(e, "Could not create key managers");
+			}
+		} else {
+			log.info("Not sending client certificate");
+		}
+
+		if (trustAll) {
+			trustManagers = getDummyTrustManager();
+		} else {
+			try {
+				trustManagers = getTrustManagers();
+			} catch (CertificateException | IOException | KeyStoreException | NoSuchAlgorithmException e) {
+				rethrow(e, "Could not create trust managers");
+			}
+		}
+		SSLContext sslCtx = null;
+
+		try {
+			sslCtx = SSLContext.getInstance("TLS");
+			sslCtx.init(keyManagers, trustManagers, null);
+		} catch (NoSuchAlgorithmException | KeyManagementException e) {
+			rethrow(e, "Could not create SSL context");
+		}
+
+		Builder builder = new OkHttpClient.Builder();
+		builder.hostnameVerifier((hostName, sslSession) -> true);
+		return builder.sslSocketFactory(sslCtx.getSocketFactory(), (X509TrustManager) trustManagers[0])
+			.build();
+	}
+
+	private static KeyManager[] getKeyManagersPem()
+		throws KeyStoreException, NoSuchAlgorithmException, UnrecoverableKeyException, IOException, CertificateException {
+		char[] randomKeyStorePass = UUIDUtil.randomUUID().toCharArray();
+		CertificateFactory certificateFactory = CertificateFactory.getInstance("X.509");
+		X509Certificate clientCert = (X509Certificate) certificateFactory.generateCertificate(new FileInputStream(CLIENT_CERT_PEM));
+		KeyStore keyStore = KeyStore.getInstance(KeyStore.getDefaultType());
+		KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+		String principal = clientCert.getSubjectX500Principal().getName();
+
+		try (PEMParser pemParser = new PEMParser(new FileReader(CLIENT_KEY_PEM))) {
+			PrivateKey privateKey = new JcaPEMKeyConverter().getPrivateKey((PrivateKeyInfo) pemParser.readObject());
+			keyStore.load(null);
```

Comment:
What is the purpose of this?

Reply:
Those are the loading parameters. There is no method without any parameters. Thus this option is used. JavaDoc of `load` allows null in this case.